content/articles/creating-a-hangman-game-with-vanilla-javascript/index.md

```diff
@@ -0,0 +1,296 @@
+
+```javascript
+var socialMedia = [
+  "whatsapp",
+  "instagram",
+  "twitter",
+  "snapchat",
+  "tiktok",
+  "youtube",
+  "wechat",
+  "facebook",
+  "telegram",
+  "bbm",
+  "palmchat",
+  "gmail",
+  "2go",
+  "linkedin",
+  "reddit",
+  "hangouts",
+  "qq",
+  "twitch",
+  "douyin"
+]
+```
+ 
+These words are going to be randomized in the game.
+
+Next, you have to define the IDs you created. Here is the code: 
+ 
+```javascript
+let answer = '';
+let maxWrong = 6;
+let mistakes = 0;
+let guessed = [];
+```
+
+The `answer` id contains the correct word to be guessed. The `maxWrong` id contains the maximum number of times you can guess wrong. The `mistakes` id contains the number of times you have guessed wrong so far, while the `guessed` id contains the letters you will guess.
+
+### The game functions
+#### The randomWord() function
+This function will help to randomly pick any of the words you placed in your social media array, which now makes the game more interesting as you do not know what word you are guessing next. Here is how the code looks like:
+
+```javascript
+function randomWord() {
+  answer = socialMedia[Math.floor(Math.random() * socialMedia.length)];
+}
+ 
+Then you call the function by using the code below:
+randomWord();
+```
+
+### The generateButtons() function
+If you notice, there are alphabet buttons in our HTML. Without them, how are we supposed to guess the words? 
+
+Well, this function helps us generate those buttons. This is a better method of creating these buttons rather than creating 26 different button tags in the HTML file. It is a good representation of "don’t-repeat-yourself" coding technique. Here's the code:
+
+```javascript
+function generateButtons() {
+  let buttonsHTML = 'abcdefghijklmnopqrstuvwxyz'.split('').map(letter =>
+    `
+      <button
+        class="btn btn-lg btn-primary m-2"
+        id='` + letter + `'
+        onClick="handleGuess('` + letter + `')"
+      >
+        ` + letter + `
+      </button>
+    `).join('');
+ 
+  document.getElementById('keyboard').innerHTML = buttonsHTML;
+}
+```
+
+In the code above, we defined all 26 letter buttons while styling them with Bootstrap. "letter" in the code represents each of the looped items in the array of alphabets that we just created in the variable. You then connect it to your HTML using the `document.getElementById('keyboard').innerHTML = buttonsHTML`. Remember, we defined the `keyboard` id in the HTML. Also, `.join('')` was added to eliminate the commas between the alphabet buttons.
+ 
+Then you call the above function with the code below:
+ 
+```javascript
+generateButtons();
+```
+ 
+Here's what the game looks like at this point:
+![hangman 1](/engineering-education/creating-a-hangman-game-with-vanilla-javascript/hangman 1.jpg)
+
```

Comment:
@Nomzy-kush Please update your images. Rename your images to remove the spaces in the filenames. Also, use descriptive filenames and alt text.

Also, check the [files changed tab](https://github.com/section-engineering-education/engineering-education/pull/2810/files) and delete the unneccesary files present in this pull request. eg. [Desktop - Shortcut.lnk](https://github.com/section-engineering-education/engineering-education/pull/2810/files#diff-e95bad44dda4cd248957949b910cb11e1285ad0f134b311cb51365ae4084b641). Please make sure you have only the files related to your article.

I also encountered some issues while doing a prelim review on your article. Be sure to check the latest commit I made to this pull request to avoid similar issues next time.